HousingFinanceInterimApi/V1/UseCase/GenerateRentPositionUseCase.cs

```diff
@@ -90,6 +90,24 @@ public async Task<StepResponse> ExecuteAsync()
                 {
                     var isSuccess = await _googleClientService.UploadCsvFile(rentPosition, $"{DateTime.Now:yyyyMMdd_HHmmss}.csv",
                         googleFileSetting.GoogleIdentifier).ConfigureAwait(false);
+                    var folderFiles = await _googleClientService.GetFilesInDriveAsync(googleFileSetting.GoogleIdentifier)
+                        .ConfigureAwait(false);
+
+
+                    LoggingHandler.LogInfo("Deleting old files");
+                    foreach (var file in folderFiles.Where(f => f.CreatedTime <= DateTime.Today.AddDays(-7)).ToList())
```

Comment:
Might not need the `.ToList()` here, as folderFiles is already returning an enumeration.
`ToList()` will iterate and append each iterated item into a new list. This has the effect of creating a temporary list which consumes memory.

Reply:
Thanks Faisal, removed the ToList in https://github.com/LBHackney-IT/housing-finance-interim-api/pull/103/commits/e84b360b7c2a109208fd4c16f431835a240b8ccc and tests still passed fine